torch/distributed/pipelining/stage.py

```diff
@@ -1434,9 +1457,13 @@ def _validate_stage_shapes(pipeline_stages: List[PipelineStage]):
             _create_metadata_tensor(device=stage.device)
             for _ in range(stage.group_size)
         ]
-        expected_outputs = stage.outputs
-        stage_output = _create_metadata_tensor(expected_outputs, device=stage.device)
-        dist.all_gather(tensor_list, stage_output)
+        outputs_meta = stage.get_outputs_meta()
+        # TODO, (1) are we deleting output validation when we move to shape inference?
+        # (2) if not, we should support multiple outputs
+        assert (
+            len(outputs_meta) == 1
+        ), f"validation logic assumes single output, got {len(outputs_meta)} outputs "
```

Comment:
Oh then should we disable validation logic (which an add-on protection) till it supports the multi-output case? Multi-output is pretty common. This assert may break a few tracer cases.

Reply:
if this is a new regression, i can remove the assert, but i thought the assert would just make another error more explicit.  The old code assumes 'output' is a single value and tries to validate against it.  Lets see if CI fails and if not, make another PR to fix validation?

Reply:
I see. Yep, if it is not a new assert, then let's see what CI says. :)